### ISSUE TITLE
sopssecretgenerator fails with "cannot parse time" error

### ISSUE THREAD

**Issue by 0anton** [goabout/kustomize-sopssecretgenerator#10]
I am trying to repeat an sample usage using latest version of sops (3.5.0), kustomize (3.5.5) and sopssecretgenerator (1.2.1) as in the documentation and getting an error:

```
kustomize build --enable_alpha_plugins
Error: env source secret-vars.env: parsing time "'2020-05-17T21:20:21Z'" as "2006-01-02T15:04:05Z07:00": cannot parse "'2020-05-17T21:20:21Z'" as "2006"
Error: failure in plugin configured via /tmp/kust-plugin-config-930760837; exit status 2: exit status 2
```
Content of the secret-vars.env file:
```
OO='ENC[AES256_GCM,data:0CaULjX4,iv:8i3cm8cCpOEimUNOa6aW6pkxkqHTrP7ymV/tLPp5q8Y=,tag:e10mL6xFXH0P46K4kraDAw==,type:str]'
sops_gcp_kms__list_0__map_resource_id='x'
sops_lastmodified='2020-05-17T21:20:21Z'
sops_mac='ENC[AES256_GCM,data:pdHugxWCYZNcVdLJdkzcWMY2B6WQ6Q/4GgR5K6NjUBSwLPS1UOze1TuQmR0smO8Za0+6YViPV2Eg8W8t7PYph0vKiODeQQr5HPC0PX0BRmobqmR/O354FiRAfxo2exsjb8bYFJNBZ17Htkjw3Jv8NKZNGwRsVhyU9A9ookYLHz4=,iv:N/DPhdSXlKJgDj/LooNlab+qVG+S1hdl07GWePb9oBI=,tag:yXIlosXaAMM1iCnj8+QLBQ==,type:str]'
sops_version='3.5.0'
sops_gcp_kms__list_0__map_enc='CiQA8Xd3GyUr1Vsb5VrVkiTSsv79QPa1JXkKVytnEj/eNe5UdJ8SYyphChQKDNHPdgwAXTHYR0EXmhD/4banCxIvCiedhjj47bL4sUzrLdUpsPwruyDqQJzvbYVlSG5oEqt+ddkEFq6zC9MQw8TWgQ8aGAoQnkI2PdBhlP/bdrJb7GX98RC/v5imDQ=='
sops_gcp_kms__list_0__map_created_at='2020-05-17T21:20:21Z'
sops_unencrypted_suffix='_unencrypted'
```

Is it a bug?

**Comment by jcassee** [comment]
Can you try to decrypt `secret-vars.env` using sops directly to see if that succeeds?
```
sops -d secret-vars.env
```

Also, do you have the source of (the relevant section of) your kustomization and your generator?

**Comment by 0anton** [comment]
Hi Joost,

Thank you for fast response. Your hint is absolutely right: sops could not decode his own format.

This error was coming from sops:

```
$ sops -d secret-vars.env
parsing time "'2020-05-17T21:20:21Z'" as "2006-01-02T15:04:05Z07:00": cannot parse "'2020-05-17T21:20:21Z'" as "2006"
```


For some reason sops surrounded each literal with single quotes and could not decode his own code:

```
$ echo FOO=secret >secret-vars.env
$ sops -e -i secret-vars.env
$ cat secret-vars.env


OO='ENC[AES256_GCM,data:0CaULjX4,iv:8i3cm8cCpOEimUNOa6aW6pkxkqHTrP7ymV/tLPp5q8Y=,tag:e10mL6xFXH0P46K4kraDAw==,type:str]'
sops_gcp_kms__list_0__map_resource_id='projects/sap-clm-clmcloud/locations/europe/keyRings/UpgradeReports/cryptoKeys/config'
sops_lastmodified='2020-05-17T21:20:21Z'
sops_mac='ENC[AES256_GCM,data:pdHugxWCYZNcVdLJdkzcWMY2B6WQ6Q/4GgR5K6NjUBSwLPS1UOze1TuQmR0smO8Za0+6YViPV2Eg8W8t7PYph0vKiODeQQr5HPC0PX0BRmobqmR/O354FiRAfxo2exsjb8bYFJNBZ17Htkjw3Jv8NKZNGwRsVhyU9A9ookYLHz4=,iv:N/DPhdSXlKJgDj/LooNlab+qVG+S1hdl07GWePb9oBI=
sops_version='3.5.0'
sops_gcp_kms__list_0__map_enc='CiQA8Xd3GyUr1Vsb5VrVkiTSsv79QPa1JXkKVytnEj/eNe5UdJ8SYyphChQKDNHPdgwAXTHYR0EXmhD/4banCxIvCiedhjj47bL4sUzrLdUpsPwruyDqQJzvbYVlSG5oEqt+ddkEFq6zC9MQw8TWgQ8aGAoQnkI2PdBhlP/bdrJb7GX98RC/v5imDQ=='
sops_gcp_kms__list_0__map_created_at='2020-05-17T21:20:21Z'
sops_unencrypted_suffix='_unencrypted'
```

It is very weird and definitily not a problem of *sopssecretgenerator*. The issue is therefore resolved for me.

As a suggestion: may be for clarity you can mark the error text, which is coming from sops? This will make the troubleshooting easier. In current output I had no clue that the error is coming from sops.

E.g.

```
kustomize build --enable_alpha_plugins
Error: env source secret-vars.env: exec of "sops -d secret-vars.env" returned : parsing time "'2020-05-17T21:20:21Z'" as "2006-01-02T15:04:05Z07:00": cannot parse "'2020-05-17T21:20:21Z'" as "2006"
Error: failure in plugin configured via /tmp/kust-plugin-config-930760837; exit status 2: exit status 2
```

**Comment by jcassee** [comment]
Weird indeed. I cannot reproduce those extra quotes with my installation of sops.

The plugin does not use the sops binary, but call the decryption code directly. However, I have added an error wrapper that notes when the error is coming from sops in 14af871237043d5889d452167eb5eaea78c20917. Hope that helps!

**Comment by jcassee** [comment]
Released as 1.2.2
